src/VisualStudio/Core/Def/Implementation/LanguageClient/AlwaysActivateInProcLanguageClient.cs

```diff
@@ -46,17 +46,24 @@ public override string Name
             => ServicesVSResources.CSharp_Visual_Basic_Language_Server_Client;
 
         protected internal override VSServerCapabilities GetCapabilities()
-            => new VSServerCapabilities
+        {
+            var isPullDiagnostics = this.Workspace.IsPullDiagnostics(InternalDiagnosticsOptions.NormalDiagnosticMode);
+            return new VSServerCapabilities
             {
                 TextDocumentSync = new TextDocumentSyncOptions
                 {
-                    Change = TextDocumentSyncKind.Incremental,
-                    OpenClose = true,
+                    // If pull diagnostics isn't enabled, then we don't really need text sync so can save a bunch of
+                    // requests that the platform would otherwise make, and state to store all of the text documents.
+                    // Strictly speaking workspace symbols could be not 100% accurate as a result, but the trade off
```

Comment:
can you clarify what you mean by: Strictly speaking workspace symbols could be not 100% accurate as a result

?

Reply:
Just that by turning off text sync, you get back the problems that it solves, namely that the LSP requests will work on whatever text is in the buffer, rather that being ordered correctly with each individual edit.

For local editing this wouldn't be a problem, but for codespaces in theory the LSP request for a search could come in before the buffer has been updated by liveshare. In practice as far as I know its impossible, and would only manifest if someone were to type in a member name and then immediately use VS Search to try to find it.

Reply:
I see. So it opens up a small period of time when you can get stale results, but things eventually catch up?